stripped/colabs/Klasyfikacja_drzewa_decyzyjne_stripped.py

```diff
@@ -188,45 +164,33 @@
 * **SoilM_100_200cm** - wilgotnosc gleby w warstwie o glebokosci od 100 do 200 cm
 """
 
-X_202210 = df_202210.loc[:,'Rainf':'SoilM_100_200cm']
-X_202211 = df_202211.loc[:,'Rainf':'SoilM_100_200cm']
-X_202212 = df_202212.loc[:,'Rainf':'SoilM_100_200cm']
-X_202301 = df_202301.loc[:,'Rainf':'SoilM_100_200cm']
-X_202302 = df_202302.loc[:,'Rainf':'SoilM_100_200cm']
-X_202303 = df_202303.loc[:,'Rainf':'SoilM_100_200cm']
-X_202304 = df_202304.loc[:,'Rainf':'SoilM_100_200cm']
-X_202305 = df_202305.loc[:,'Rainf':'SoilM_100_200cm']
-X_202306 = df_202306.loc[:,'Rainf':'SoilM_100_200cm']
-X_202307 = df_202307.loc[:,'Rainf':'SoilM_100_200cm']
-X_202308 = df_202308.loc[:,'Rainf':'SoilM_100_200cm']
-X_202309 = df_202309.loc[:,'Rainf':'SoilM_100_200cm']
-
-y_m1 = df_202307['klasa']
-y_m2 = df_202307['pustynia']
-y_m3 = df_202307['step']
+y_m1 = list_of_df_all[0]['klasa']
+y_m2 = list_of_df_all[0]['pustynia']
+y_m3 = list_of_df_all[0]['step']
+
+list_of_df = [df.loc[:,'Rainf':'SoilM_100_200cm'] for df in list_of_df_all]
 
 """#### Definiowanie funkcji"""
 
-def information_gain(X: pd.DataFrame, y: pd.DataFrame) -> None:
-  importances = mutual_info_classif(X, y)
-  feature_info = pd.Series(importances, X.columns).sort_values(ascending=False)
-  print(feature_info)
-
-def correlations(X: pd.DataFrame) -> None:
-  cor = X.corr()
-  sns.heatmap(cor, annot=True,cmap='Reds')
-
-def plot_data_dist(y: pd.DataFrame) -> None:
-  dane = pd.Series(y).value_counts().sort_index()
-  labels = list(np.sort(pd.unique(y)))
-  ypos=np.arange(len(labels))
-  plt.xticks(ypos, labels)
-  plt.xlabel('Klasa')
-  plt.ylabel('Czestosc')
-  plt.title('Liczebnosc dla proby')
-  plt.bar(ypos,dane)
-
-def summary_model(model, X:pd.DataFrame, y:pd.DataFrame, labels_names: List) -> None:
+def create_models(params_gs: dict, list_of_df: List[pd.DataFrame], scoring: Optional[str] = 'accuracy') -> tuple[List, List]:
+  list_params = []
+  list_of_models = []
+  number_of_data = len(list_of_df)
+
+  for i in range(number_of_data):
+    gs = GridSearchCV(tree.DecisionTreeClassifier(random_state = 2024), cv = 5, param_grid = params_gs, scoring = scoring)
+    gs.fit(list_of_df[i][0], list_of_df[i][1])
+    list_params.append(gs.best_params_)
+
+  for i in range(number_of_data):
+    list_of_models.append(tree.DecisionTreeClassifier(random_state = 2024, **list_params[i]))
+
+  for i in range(number_of_data):
+    list_of_models[i].fit(list_of_df[i][0], list_of_df[i][1])
```

Comment:
3 kolejne pętle po tym samym zakresie. Może da się to zrobić jedną pętlą. Jeśli nie - poproszę o komentarz czemu są tu 3 pętle.

Reply:
Zrobiono jedną pętlą